mkosi/__init__.py

```diff
@@ -249,27 +249,6 @@ def prepare_tree_root(state: MkosiState) -> None:
             btrfs_subvol_create(state.root)
 
 
-def prepare_tree(state: MkosiState, cached: bool) -> None:
-    if cached:
-        # Reuse machine-id from cached image.
-        state.machine_id = uuid.UUID(state.root.joinpath("etc/machine-id").read_text().strip()).hex
-        # Always update kernel command line.
-        if not state.do_run_build_script and state.config.bootable:
-            state.root.joinpath("etc/kernel/cmdline").write_text(" ".join(state.config.kernel_command_line) + "\n")
-        return
-
-    with complete_step("Setting up basic OS tree…"):
-        state.root.mkdir(mode=0o755, exist_ok=True)
-        # We need an initialized machine ID for the build & boot logic to work
-        state.root.joinpath("etc").mkdir(mode=0o755, exist_ok=True)
-        state.root.joinpath("etc/machine-id").write_text(f"{state.machine_id}\n")
-
-        state.root.joinpath("etc/kernel").mkdir(mode=0o755, exist_ok=True)
-        state.root.joinpath("etc/kernel/cmdline").write_text(" ".join(state.config.kernel_command_line) + "\n")
-        state.root.joinpath("etc/kernel/entry-token").write_text(f"{state.machine_id}\n")
-        state.root.joinpath("etc/kernel/install.conf").write_text("layout=bls\n")
```

Comment:
not that it matters terribly, since we don't guarantee that the images can be updated, but since we generate UKIs wouldn't the proper way to specify `layout=uki`. Probably moot, though, since with v254 this will probably be autodetected.

Reply:
Ideally we could set the layout via the CLI of bootctl and kernel-install, but we can't. I want to write the least amount of files possible in the image itself, so I'd prefer to not configure the layout actually

Reply:
For `kernel-install` we could do it via environment variable (`KERNEL_INSTALL_LAYOUT=uki`), but yes, for `bootctl` nothing like this exists to my knowledge.

Reply:
I'm pretty sure that doesn't work, kernel-install only supports reading the layout from the config file

Reply:
Yep, looking through the code, you're right.